build.py

```diff
@@ -834,12 +842,46 @@ def TEMPFILE(cmd_, comstr=None):
             if list_targets[0].startswith("@"):
                 test_name = list_targets[0]
                 test_name = '+' + test_name[1:]
+
+                # These take priority over unit test shortcut names.
+                self.unittest_skipped_shortcuts.add(test_name)
+                if test_name in self.unittest_shortcuts.keys():
+                    del self.unittest_shortcuts[test_name]
+
                 self.builds.append(dict(
                         rule='RUN_TEST',
                         outputs=test_name,
                         inputs=list_sources
                     ))
 
+                # Add shortcuts for this unit test.
+                for unit_test_source_file in \
+                        n.executor.get_all_children()[0].executor.get_all_sources():
+                    # Get the file name
```

Comment:
I suggest using python's `os.path` to perform these semantic operations. I think you want  `splitext(basename(file))[0]`

Reply:
That's a neat python module, done!